include/mcl_3dl/state_6dof.h

```diff
@@ -191,27 +194,27 @@ class State6DOF : public mcl_3dl::pf::ParticleBase<float>
       p.z = t.z_;
     }
   }
-  static State6DOF generateNoise(
-      std::default_random_engine& engine_,
-      const State6DOF& mean, const State6DOF& sigma)
+  template <typename T, typename RANDOM_ENGINE, typename NOISE_GEN>
+  static State6DOF generateNoise(RANDOM_ENGINE& engine, const NOISE_GEN& gen)
   {
-    State6DOF noise;
-    if (mean.isDiff() || !sigma.isDiff())
+    if (gen.getDimension() != 6)
     {
-      ROS_ERROR("Failed to generate noise. mean must be mcl_3dl::Quat and sigma must be rpy vec.");
+      ROS_ERROR("Dimension of noise must be 6. Passed: %lu", gen.getDimension());
     }
+    State6DOF noise;
+    const auto org_noise = gen(engine);
```

Comment:
This auto would better be specified explicitly to avoid confusing with State6DOF type.

Reply:
Done